### PR TITLE
Adds the option to do a count distinct query

### PR DESCRIPTION
**Pull request by parkerl**
- allows the atom ":distinct" to be passed to the count macro
  - implements count distinct for postgres and mysql
  - issue #874

### CODE REVIEW

File: lib/ecto/query/builder.ex [hunk 1]
```diff
@@ -1,6 +1,8 @@
 defmodule Ecto.Query.Builder do
   @moduledoc false
 
+  @aggregators ~w(count)a
```
Comment (josevalim) [inline lib/ecto/query/builder.ex]: Sorry, lets call it `@distinct`.

**Comment by josevalim** [comment]
@parkerl we just need docs, as you said. They can be added to Ecto.Query.API. :)

**Comment by parkerl** [comment]
@josevalim like that?

**Comment by parkerl** [comment]
@josevalim sorry I missed your comment earlier. Also found another bit of cruft.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by josevalim** [comment]
Thank you for improving Ecto!

**Pull request merged**